### ISSUE TITLE
Document about different ways of login

### ISSUE THREAD

**Issue by kimyen** [Sage-Bionetworks/synapseDocs#422]
1. There is a docs for `.synapseConfig` file, but not explicitly talking about how to use it for long term access to Synapse (including using apiKey).

2. `syn.login()` with `rememberMe=TRUE` on Mac will cause keychain to prompt user to allow the Synapse client to access the keychain password. User may select "No" which later, they cannot login using the password saved in keychain. 

These topics should be documented once and referenced in the Python client and synapser client docs.

**Comment by kimyen** [comment]
Use case from @apratap:
Using `.synapseConfig` file is very useful for users who would share their codes with others. When one shares a chunk of code that uses Synapse client, in the code, they should only use `syn.login()` without putting in their username and password. However, there isn't a formal documentation to point others to setup their `.synapseConfig` file in order to run the shared code.

**Comment by kimyen** [comment]
I think we should write the docs in term of use cases:
1. One time login
Use `username` and `password` as follow:
Python client:
```
import synapseclient
syn = synapseclient.login('username', 'password')
```
synapser client:
```
library("synapser")
synLogin("username", "password")
```
2. For writing code and share with others
For writing code using Python/ synapser client, that easy to share with others, one does not include their credentials in the code. Instead, user `.synapseConfig` file. You can prefer this documentation to your collaborators on how to setup `.synapseConfig` file.
A. The `.synapseConfig` file
For Python client:
The `.synapseConfig` file is automatically installed.
For synapser client, installing `.synapseConfig` at installation. For now, please create a `.synapseConfig` file in your home directory.
B. Add your credentials to `.synapseConfig` file:
For users who have a default `.synapseConfig` file, look for the section below and add your credentials. Notes that they may be commented out. You would need to un-comment these lines.
For users who created an empty `.synapseConfig` file, add the below section to your `.synapseConfig` file:
```
[authentication]
username = your_user_name
password = your_password
```
C. Login in the client
Python client:
```
import synapseclient
syn = synapseclient.login()
```
synapser client:
```
library("synapser")
synLogin()
```
3. Letting the OS manage credentials 
For users who would like to save their credentials and let other OS configured application (like keychain in Mac) manages credentials for you, when login the first time, use:
In Python:
```
import synapseclient
syn = synapseclient.login("username", "password", rememberMe=True)
```
In synapser:
```
library("synapser")
synLogin("username", "password", rememberMe=True)
```
Then the application (keychain in Mac) will prompt you to allow other applications including Python and R to access this credentials. Please choose "Yes" or "OK".
The second time you login, you will not have to enter your username and password:
Python client:
```
import synapseclient
syn = synapseclient.login()
```
synapser client:
```
library("synapser")
synLogin()
```

**Comment by kdaily** [comment]
I like it! It mirrors what docker does in many ways (I was reminded by this when logging into docker and having it save my credentials in plain text - but then it didn't keep complaining about it, which I appreciate):

https://docs.docker.com/engine/reference/commandline/login/#privileged-user-requirement

**Comment by kdaily** [comment]
Also you forgot the command line client!

```
synapse -c /path/to/synapseConfig
synapse --configPath /path/to/synapseConfig
```

**Comment by kelshmo** [comment]
filed a [Jira](https://sagebionetworks.jira.com/browse/SYNR-1367) and working with Kim to get better R/python client documentation before pulling this into the synapse docs

**Comment by kimyen** [comment]
r-docs: https://github.com/Sage-Bionetworks/synapser/pull/216

**Comment by kimyen** [comment]
@tschaffter also suggested [SYNPY-922](https://sagebionetworks.jira.com/browse/SYNPY-922), which is related to this discussion. 

The Python docs will be available in version 1.9.2.

**Comment by kimyen** [comment]
This is done.
